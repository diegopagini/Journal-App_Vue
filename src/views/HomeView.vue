<!-- @format -->

<template>
	<div class="home">
		<h1>Hola Mundo</h1>
		<button
			@click="goToDayBook"
			class="btn btn-primary"
		>
			Primary
		</button>
	</div>
</template>

<script>
	export default {
		methods: {
			goToDayBook() {
				this.$router.push({ name: 'daybook' });
			},
		},
	};
</script>
