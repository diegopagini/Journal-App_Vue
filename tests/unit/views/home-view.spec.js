/** @format */
import HomeView from '@/views/HomeView.vue';
import { shallowMount } from '@vue/test-utils';

describe('HomeView', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallowMount(HomeView);
	});

	it('Should render the component', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('Click a button should redirect to daybook', () => {
		const mockRouter = {
			push: jest.fn(),
		};

		const wrapper = shallowMount(HomeView, {
			global: {
				mocks: {
					$router: mockRouter,
				},
			},
		});

		wrapper.find('button').trigger('click');
		expect(mockRouter.push).toHaveBeenCalled();
		expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });
	});
});
