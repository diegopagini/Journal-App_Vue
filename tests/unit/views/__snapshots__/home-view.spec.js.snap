// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView Should render the component 1`] = `
<div class="home">
  <h1>Hola Mundo</h1><button class="btn btn-primary"> Primary </button>
</div>
`;
